File: Examples/DotNET/VisualBasic/Document/SignPdfDoc.vb
Imports Com.Aspose.PDF.Api
Imports Com.Aspose.PDF.Model
Imports Com.Aspose.Storage.Api

Namespace Document
    Class SignPdfDoc
        Public Shared Sub Run()
            ' ExStart:1
            Dim pdfApi As New PdfApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH)
            Dim storageApi As New StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH)

            Dim fileName As [String] = "sample-input.pdf"
            Dim signatureFileName As [String] = "pkc7-sample.pfx"
            Dim pageNumber As Integer = 1
            Dim storage As [String] = ""
            Dim folder As [String] = ""

            Dim body As New Signature()
            body.Authority = "Naeem Akram"
            body.Location = "Gojra"
            body.Contact = "naeem.akram@aspose.com"
            body.[Date] = "06/24/2015 2:00:00.000 AM"
            body.FormFieldName = "Signature1"
            body.Password = "aspose"
            Dim rect As New Rectangle()
            rect.X = 100
            rect.Y = 100
            rect.Height = 100
            rect.Width = 200
            body.Rectangle = rect
            body.SignaturePath = signatureFileName
            body.SignatureType = "PKCS7"
            body.Visible = True

            Try
                ' Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName))
                storageApi.PutCreate(signatureFileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + signatureFileName))

                ' Invoke Aspose.PDF Cloud SDK API to sign Pdf document
                Dim apiResponse As SaaSposeResponse = pdfApi.PostSignPage(fileName, pageNumber, storage, folder, body)

                If apiResponse IsNot Nothing AndAlso apiResponse.Status.Equals("OK") Then
                    Console.WriteLine("Sign PDF Documents, Done!")
                    Console.ReadKey()
                End If
            Catch ex As Exception
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + vbLf + ex.StackTrace)
            End Try
            ' ExEnd:1
        End Sub
    End Class
End Namespace

File: Examples/Node.js/Document/SignPdfDoc.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var PdfApi = require('asposepdfcloud');
var StorageApi = require('asposestoragecloud');
var configProps = require('../config/config.json');
var data_path = '../../../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose.Pdf API SDK
var pdfApi = new PdfApi(config);

var name = "sample-input-2.pdf";
var signatureFileName = "pkc7-sample.pfx";
	
var signatureBody = {
			'Authority' : 'Farooq Sheikh',
			'Location' : 'Rawalpindi',
			'Contact' : 'farooq.sheikh@aspose.com',
			'Date' : '06/24/2015 2:00:00.000 AM',
			'FormFieldName' :  'Signature1',
			'Password' : 'aspose',
			'SignaturePath' : signatureFileName,
			'SignatureType' : 'PKCS7',
			'Visible' : true,
			'Rectangle' : {
				'X' : 100,
				'Y' : 100,
				'Height' : 100,
				'Width' : 200
			}
	};

try {
// Upload file to aspose cloud storage
storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {	
	assert.equal(responseMessage.status, 'OK');

	storageApi.PutCreate(signatureFileName, null, null, data_path + signatureFileName , function(responseMessage) {	
		assert.equal(responseMessage.status, 'OK');

		// Invoke Aspose.Pdf Cloud SDK API to sign PDF document
		pdfApi.PostSignDocument(name, null, null, signatureBody, function(responseMessage) {
				assert.equal(responseMessage.status, 'OK');

				// Download signed pdf from storage server
				storageApi.GetDownload(name, null, null, function(responseMessage) {
					assert.equal(responseMessage.status, 'OK');
					var writeStream = fs.createWriteStream(data_path + "sample-input-2_out.pdf");
					writeStream.write(responseMessage.body);
					});
			});
		});
	});

}
catch (e) 
{
  console.log("exception in example");
  console.log(e);
}
//ExEnd:1